ionoscloud/resource_server_boot_device_selection.go

```diff
@@ -56,23 +56,34 @@ func resourceServerBootDeviceSelection() *schema.Resource {
 }
 
 func resourceServerBootDeviceSelectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	ss := cloudapiserver.Service{Client: meta.(services.SdkBundle).CloudApiClient, Meta: meta, D: d}
 	dcId := d.Get("datacenter_id").(string)
 	serverId := d.Get("server_id").(string)
-	bootDeviceId := d.Get("boot_device_id").(string)
+
+	ss := cloudapiserver.NewUnboundService(serverId, meta)
 
 	// The bootable device to which the server will revert if this resource is destroyed.
 	defaultBootVolume, err := ss.GetDefaultBootVolume(ctx, dcId, serverId)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	if err := ss.UpdateBootDevice(ctx, dcId, serverId, bootDeviceId); err != nil {
-		return diag.FromErr(err)
+	if err = d.Set("default_boot_volume_id", defaultBootVolume.Id); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting a default boot volume for boot selection resource"))
+	}
+
+	bootDeviceIDValue, bootDeviceIDOk := d.GetOk("boot_device_id")
+	if !bootDeviceIDOk {
+		if err := ss.PxeBoot(ctx, dcId, serverId); err != nil {
+			return diag.FromErr(fmt.Errorf("error while performing pxe boot: %s, serverId: %s, dcId: %s", err.Error(), serverId, dcId))
```

Comment:
error is usually the last and with %w. Any reason to have it this way?

Reply:
fixed to correctly wrap error ([555645e](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/521/commits/555645efa0ea6f7cdbb227cfb1aae3fb448e341d))